File: app/src/main/java/com/hqq/example/ui/jetpack/package1/MvvmTestActivity.java
package com.hqq.example.ui.jetpack.package1;

import android.app.Activity;
import android.content.Intent;

import androidx.databinding.ObservableArrayList;
import androidx.databinding.ObservableArrayMap;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import com.hqq.core.ui.vm.BaseActivity;
import com.hqq.example.R;
import com.hqq.example.databinding.ActivityMvvmBinding;
import com.hqq.example.ui.jetpack.livedata.User;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.example.ui.jetpack.package1
 * @FileName :   Package1Activity
 * @Date : 2020/7/21 0021  下午 3:53
 * @Email :  qiqiang213@gmail.com
 * @Descrive : ViewBinding + ViewModel + liveData
 */
public class MvvmTestActivity extends BaseActivity<ActivityMvvmBinding,UserViewModel> {
    public static void open(Activity context) {
        Intent starter = new Intent(context, MvvmTestActivity.class);
        context.startActivityForResult(starter, -1);
    }

    @Override
    protected void addViewModel() {
//       mBinding.setVariable(BR.vm, new ViewModelProvider(this).get(UserViewModel.class));
        mBinding.setVm(new ViewModelProvider(this).get(UserViewModel.class));
        mBinding.setTitle("这是一个变量String");
        ObservableArrayMap<String, String> map = new ObservableArrayMap<>();
        map.put("key1", "ObservableArrayMap  value1");
        map.put("key2", "ObservableArrayMap value2");
        map.put("key3", "ObservableArrayMap value3");
        map.put("key4", "ObservableArrayMap value4");
        mBinding.setMap(map);

        ObservableArrayList observableArrayList = new ObservableArrayList<String>();
        observableArrayList.add("key1");
        observableArrayList.add("key2");
        mBinding.setList(observableArrayList);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_mvvm;
    }

    @Override
    protected void initViews() {

        mBinding.getVm().getLiveData().observeForever(new Observer<User>() {
            @Override
            public void onChanged(User user) {

            }
        });

    }

   public static final class Fields {
        public final static MutableLiveData<Integer> AGE = new MutableLiveData<>(1);
        public final  static int AGE2=1;
    }

}

File: app/src/main/java/com/easy/example/ui/jetpack/package1/MvvmTestActivity.kt
package com.easy.example.ui.jetpack.package1

import android.app.Activity
import android.content.Intent
import androidx.databinding.ObservableArrayList
import androidx.databinding.ObservableArrayMap
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModelProvider
import com.easy.core.ui.base.BaseVmActivity
import com.easy.core.utils.log.LogUtils.e
import com.easy.example.R
import com.easy.example.databinding.ActivityMvvmBinding

/**
 * @Author : huangqiqiang
 * @Package : com.easy.example.ui.jetpack.package1
 * @FileName :   Package1Activity
 * @Date : 2020/7/21 0021  下午 3:53
 * @Email :  qiqiang213@gmail.com
 * @Descrive : ViewBinding + ViewModel + liveData
 */
class MvvmTestActivity : BaseVmActivity<UserViewModel, ActivityMvvmBinding, >() {
    override fun addViewModel() {
        e(" MvvmTestActivity   addViewModel 1 ")
        //       mBinding.setVariable(BR.vm, new ViewModelProvider(this).get(UserViewModel.class));
        binding.vm = ViewModelProvider(this).get(UserViewModel::class.java)
        binding.title = "这是一个变量String"
        val map = ObservableArrayMap<String, String>()
        map["key1"] = "ObservableArrayMap  value1"
        map["key2"] = "ObservableArrayMap value2"
        map["key3"] = "ObservableArrayMap value3"
        map["key4"] = "ObservableArrayMap value4"
        binding.map = map
        val observableArrayList: ObservableArrayList<String> = ObservableArrayList<String>()
        observableArrayList.add("key1")
        observableArrayList.add("key2")
        binding.setList(observableArrayList)
        e(" MvvmTestActivity   addViewModel 2 ")
    }

    override val bindingViewModelId: Int
        get() = 0

    override val layoutId: Int
        get() = R.layout.activity_mvvm

    override fun initViews() {
        binding?.vm?.liveData?.observeForever { }
    }

    object Fields {
        val AGE = MutableLiveData(1)
        const val AGE2 = 1
    }

    companion object {
        fun open(context: Activity) {
            val starter = Intent(context, MvvmTestActivity::class.java)
            context.startActivityForResult(starter, -1)
        }
    }
}